subsys/logging/log_output.c

```diff
@@ -216,14 +278,25 @@ static int timestamp_print(const struct log_output *output,
 			length = print_formatted(output, "%s.%06uZ ",
 						 time_str, ms * 1000U + us);
 #else
```

Comment:
imo, this could also be converted to `if (IS_ENABLED(..))` so that ifdef around `get_YMD_from_seconds()` implementation could be removed.

Reply:
I tried to change this one as well, but of course removing the `#if defined` around the `strftime()` function causes a warning for "undefined function" when building without `CONFIG_NEWLIB_LIBC`.

My proposal in order to remove all the conditionals `#if def` was to add an `__attribute__((unused))` for the `get_YMD_from_seconds()` function. Is this acceptable?